openapi/generator.go

```diff
@@ -799,6 +813,27 @@ func (g *Generator) schemaFromComponents(schema *SchemaOrRef) *Schema {
 // of the given type, transformed to CamelCase without
 // a dot separator between the two parts.
 func (g *Generator) typeName(t reflect.Type) string {
+	if t.Kind() == reflect.Ptr {
+		t = t.Elem()
+	}
+	// If the name of the type was overidded,
+	// use it in priority.
+	if name, ok := g.typeNames[t]; ok {
+		return name
+	}
+	// At this point, we can't use the Implements() method to
+	// check if the type t implements the TypeNamer interface
+	// because any pointer has been dereferenced in this func
+	// or before and it won't match if the method implementing
+	// the interface has a pointer receiver.
+	// Instead, we create a new instance of t's type and use a
```

Comment:
Actually, it's still possible to check with `Implements` by having a pointer to your type with `reflect.PtrTo`.

Example here: https://play.golang.org/p/6H-s6UlkE64